### ISSUE TITLE
The definition source of createClient is the one that has been deprecated.

### ISSUE THREAD

**Issue by yanachuwan9sm** [contentful/contentful-management.js#1701]
Hi.
As the title suggests.
It seems to be oriented towards the deprecated source of the createClient definition, even though the second argument is not passed.👋

The following is a [following the README of contentful-typescript-codegen](https://github.com/intercom/contentful-typescript-codegen)
The code for getContentfulEnvironment.js created in the root of the project.

<img width="831" alt="image" src="https://user-images.githubusercontent.com/95360878/220226331-04fe82b5-977c-4512-8072-55a1000a6493.png">

As you can see below, is it possible that `@deprecated` is affecting the overload?
I felt that Is this a phenomenon that I am the only one reproducing? 

(If so, I felt it necessary to review the VSCode settings, etc.)

https://github.com/microsoft/tsdoc/issues/131#:~:text=%E5%95%8F%E9%A1%8C%E3%81%AB%E6%B0%97%E3%81%A5%E3%81%8D%E3%81%BE%E3%81%97%E3%81%9F%E3%80%82%40deprecated%E3%81%AF%E3%82%AA%E3%83%BC%E3%83%90%E3%83%BC%E3%83%AD%E3%83%BC%E3%83%89%E3%81%AB%E5%BD%B1%E9%9F%BF%E3%81%97%E3%81%BE%E3%81%99%E3%80%82

Also, by modifying `contentful-management.d.ts` as follows
The definition source is now oriented to the createClient, which can be expected.
(I don't think this is in line with [best practice](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#function-overloads), but... ,)

I am not very knowledgeable in this area, so please can someone lend me their knowledge🙏

```ts
/**
 * Create a client instance
 * @param params - Client initialization parameters
 *
 * ```javascript
 * const client = contentfulManagement.createClient({
 *  accessToken: 'myAccessToken'
 * })
 * ```
 */
declare function createClient(params: ClientOptions): ClientAPI;
declare function createClient(params: ClientOptions, opts: {
    type: 'plain';
    defaults?: DefaultParams;
}): PlainClientAPI;

/**
 * @deprecated
 */
export type ClientParams = RestAdapterParams & UserAgentParams;
type ClientOptions = (RestAdapterParams | AdapterParams) & UserAgentParams;
/**
 * @deprecated the alphaFeatures option is not longer supported.
 *
 * {@link Use (createClient:1) (createClient:2) } instead.
 */
declare function createClient(params: ClientOptions, opts: {
    type?: 'plain';
    alphaFeatures: string[];
    defaults?: DefaultParams;
}): ClientAPI | PlainClientAPI;

```
Versions are as follows: 
`"contentful-management": "10.29.4"`

[imo]

I also thought that `@deprecated` should include a recommendation of what should be used instead, so I thought we should follow [convention](https://github.com/Microsoft/tsdoc/blob/3f18e21693e9a0fd9f3949b1d2c2f3820d372cdd/tsdoc/src/details/StandardTags.ts#L50).

**Comment by filipposwift** [comment]
I'm having the some problem with:
`"contentful-management": "^10.34.1"`
Any fix or suggestions?

**Comment by Chaoste** [comment]
Hello everyone, thanks for bringing this up. It appears that this is a malicious error in the IDE and we found a way to mitigate this. There is a more detailed explanation in the linked PR. As recommended by you, I adjusted the warning so that is "followed by a sentence describing the recommended alternative". It doesn't say that one should link to another function and I found it not very intuitive to link to other overloads, so I decided to leave that out and describe with words what one should do.
